codice/ddf
# Remove Solr XPath support

#### What does this PR do?
Removes support for XPath queries to the Solr provider. Deprecated XPath methods on `FilterDelegate`.

#### Who is reviewing it? 

@lamhuy 
@derekwilhelm 

#### Select relevant component teams: 

@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.

@rzwiefel 
@coyotesqrl

#### How should this be tested?

Verify everything builds and the itests pass.  Verify starting the system and search still works.

#### Any background context you want to provide?
Lux is not maintained and has not been updated for newer versions of Solr. Our fork of Lux has not been updated in 5 years. Dependencies used by Lux are starting to accumulate vulnerabilities which will eventually impact system security. Prefiltering has not worked for 4 major versions of Solr now.  It is possible to create XPath queries that would need to do a brute force XPath evaluation on every document. This could become a stability and performance risk. The majority of the users of the system extract values out of metadata XML to be indexed in fields instead of using XPath queries.  Removing XPath support will reduce index size, memory usage, and document response size from Solr.  For these reasons, it is better to remove the complexity of XPath support.

https://github.com/codice/lux/commits/solr-4.10.4

#### What are the relevant tickets?

#### Screenshots


#### Checklist:
- [ n/a ] Update / Add Threat Dragon models

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
